posts/atomic-design-bootstrap.md

```diff
@@ -0,0 +1,154 @@
+
+The global idea is to define a graphic design. _Bootstrap_ being an existing Pattern Library, there may be a risk of conflict between the _Bootstrap_ choices and ours.
+
+Nevertheless, using this type of framework brings advantages (reusable components, graphic coherence, numerous contributors ...); it may be difficult to remove it.
+
+That's why I'm going to show you several examples to show you how to make _Atomic Design_ and _Bootstrap_ coexist.
+
+# Try to not use Bootstrap components
+
+First of all, I recommend you to avoid the use of _Bootstrap_ [Components](https://getbootstrap.com/docs/4.4/components). Why? Because they are graphic elements with a strong impact on what the application is doing.
+
+As an example with [buttons](https://getbootstrap.com/docs/4.4/components/buttons). They have some variants depending on their states (hover, focus, disabled…) and it gets more complicated when assembling them with other buttons.
+
+![Bootstrap button with states: normal, hover, focus, disabled](https://raw.githubusercontent.com/gnuk/blog-usa/atomic-bootstrap/images/2020/01/atomic-design-bootstrap-buttons.png)
+_Buttons states from Bootstrap website_
+
+That's why, if you choose to use _Bootstrap_, you will impact the brand image but also the mechanic of your application. Therefore, it is better to make your own buttons with your own semantic.
+
+> Yes, but they are already ready, why not reuse them? Especially since many people maintain them?
+
+This is the argument I hear systematically when we talk about this kind of tools "it exists", "why reinvent the wheel?". I will try to answer these remarks beginning with two quotes from __Alla Kholmatova__ book, [Design Systems](http://designsystemsbook.com):
+
+> __Functional Patterns__ are tangible building blocks of the interface. Their purpose is to enable or encourage certain user behaviors.
+
+_page 63_
+
+> Example of __perceptual patterns__ […] include tone of voice, typography, color palette, layouts, illustrations and iconography styles, shapes and textures, spacing, imagery, interactions or animations, and all the specific ways in which those elements are combined and used in an interface.
+
+_page 82_
+
+Thus, for our buttons, it is necessary to reflect before choosing a ready to use component:
+
+* Why do you need a blue button with rounded corners? all states? all sizes? 9 semantic variants proposed?
+* Do you need glued buttons when grouped?
+* How many time need you to change colors? add new variants? remove rounded corners? shadows?
+* What will happen if you need to upgrade Bootstrap? Even if it is a minor release?
+
+You can follow this type of remarks to other [Components](https://getbootstrap.com/docs/4.4/components/buttons) to understand it is not relevant to use them when you need a graphical identity ; because you may take a lot of time to override and maintain them.
```

Comment:
```suggestion
You can follow these types of remarks about other [Components](https://getbootstrap.com/docs/4.4/components/buttons) to understand it is not relevant to use them when you need a graphical identity ; because you may take a lot of time to override and maintain them.
```